### PR TITLE
Remove the CHERI mode enable CSR bits

### PR DESCRIPTION
**Pull request by arichardson**
With the mode bit of PCC always present when legacy mode is available,
we no longer need a separate out-of-band state bit to indicate the
current mode. The CHERI register enable (CRE) bit remains present to
prevent access to CHERI instruction and mseccfg.CME was changed to
behave as a register enable bit for M-mode.

Having the mode bit always present ensures that we can set the execution
mode for trap handlers without affecting the mode of other code.

### CODE REVIEW

**Comment by arichardson** [comment]
CC: @sorear

This relates to https://github.com/riscv/riscv-cheri/pull/130, moving all CHERI mode state to PCC and out of xenvcfg.

File: src/img/menvcfgmodereg.edn [hunk 1]
```diff
@@ -4,13 +4,12 @@
 (def row-height 45)
 (def row-header-fn nil)
 (def boxes-per-row 32)
-(draw-column-headers {:height 20 :font-size 18 :labels (reverse ["0" "" "1" "3" "4" "5" "6" "" "7" "" "8" "" "" "" "" "" "27" "28" "" "29" "" "30" "" "" "" "" "" "61" "" "62" "" "63"])})
+(draw-column-headers {:height 20 :font-size 18 :labels (reverse ["0" "" "1" "3" "4" "5" "6" "" "7" "" "8" "" "" "" "" "" "" "27" "28" "" "" "" "" "" "" "" "" "61" "" "62" "" "63"])})
 
 (draw-box "STCE" {:span 2})
 (draw-box "PBMTE" {:span 2})
-(draw-box (text "WPRI" {:font-weight "bold"}) {:span 7})
+(draw-box (text "WPRI" {:font-weight "bold"}) {:span 9})
```
Comment (tariqkurd-repo) [inline src/img/menvcfgmodereg.edn:11]: This isn't rendering correctly

image

and it's only 1 bit in the text, not 2

File: src/img/senvcfgreg.edn [hunk 2]
```diff
@@ -6,20 +6,18 @@
 (def left-margin 30)
 (def right-margin 30)
 (def boxes-per-row 32)
-(draw-column-headers {:height 20 :font-size 18 :labels (reverse ["0" "" "1" "3" "4" "5" "6" "" "7" "" "8" "" "" "" "" "" "" "" "" "" "" "" "" "27" "28" "" "29" "" "30" "" "" "SXLEN-1"])})
+(draw-column-headers {:height 20 :font-size 18 :labels (reverse ["0" "" "1" "3" "4" "5" "6" "" "7" "" "8" "" "" "" "" "" "" "" "" "" "" "" "" "" "27" "28" "" "" "" "" "" "SXLEN-1"])})
 
-(draw-box (text "WPRI" {:font-weight "bold" :font-size 20}) {:span 4})
+(draw-box (text "WPRI" {:font-weight "bold" :font-size 20}) {:span 6})
```
Comment (tariqkurd-repo) [inline src/img/senvcfgreg.edn:11]: This has the same problem as menvcfg

image

**Review by tariqkurd-repo — CHANGES_REQUESTED** [review]
two minor fixes required as listed, otherwise LGTM

**Comment by andresag01** [comment]
@tariqkurd-repo : Could you review again? I fixed the bit indexes in xenvcfg diagrams

**Comment by andresag01** [comment]
@tariqkurd-repo @arichardson : This was merged onto an alternate branch instead of main. We need to merge it into main if we are happy with the changes

**Comment by arichardson** [comment]
> @tariqkurd-repo @arichardson : This was merged onto an alternate branch instead of main. We need to merge it into main if we are happy with the changes

I think the easiest way will be to recreate this PR and merge it. Will do so shortly.

**Comment by sorear** [comment]
I will most likely not be able to look at any CHERI-related issues until two weeks from now.

**Pull request merged**